init/eessi_archdetect.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+# current pathways implemented in EESSI
+# x86_64/generic
+# x86_64/intel/haswell
+# x86_64/intel/skylake_avx512
+# x86_64/amd/zen2
+# x86_64/amd/zen3
+# aarch64/generic
+# ppc64le/generic
+# ppc64le/power9le
+
+ARGUMENT=${1:-none}
+
+cpupath () {
+  # let the kernel tell base machine type
+  MACHINE_TYPE=${EESSI_MACHINE_TYPE:-$(uname -m)}
+
+  # fallback path
+  CPU_PATH="${MACHINE_TYPE}/generic"
+
+  if [ ${MACHINE_TYPE} == "aarch64" ]; then
+    echo ${CPU_PATH}
+    exit
+  fi
+
+  if [ ${MACHINE_TYPE} == "ppc64le" ]; then
+    PROC_CPUINFO=${EESSI_PROC_CPUINFO:-/proc/cpuinfo}
```

Comment:
This should move up, to only have it once

Reply:
done